app/services/school_record/enrol.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class SchoolRecord::Enrol < BaseService
+  def call
+    ActiveRecord::Base.transaction do
+      if leaving_date
+        # update previous school record. Might need to pass school from/to
+        previous_school_record.update(leaving_date:)
+      end
+
+      participant_profile.school_records.create!(
+        school:,
+        joining_date:,
+      )
+    end
+  end
+
+private
+
+  attr_reader :participant_profile, :school, :joining_date, :leaving_date
+
+  def initialize(participant_profile:, school:, joining_date:, leaving_date: nil)
+    @participant_profile = participant_profile
+    @school = school
+    @joining_date = joining_date
+    @leaving_date = leaving_date
+  end
+
+  def previous_school_record
+    # participant_profile.school_records.latest
```

Comment:
we might need more logic around determining latest school record: 
no leaving date, or pass in previous school etc. as well as look at timestamp

Reply:
from @peteryates: For mentors, this is more complex as a participant can belong to many schools at once, how do we find the correct school record?

Reply:
We should not mix up mentors training and mentors mentoring. They should be two separate things.

Reply:
> from @peteryates: For mentors, this is more complex as a participant can belong to many schools at once, how do we find the correct school record?

Mentors receive training only from one school, so we should be fine.

They can also mentor ECTs in multiple schools via the school_mentor association, but like Matt mentioned above, it's a separate thing.

Reply:
I didn't even realise that's what the `schools_mentors` table meant! 🤦🏽